docs/build_docs.py

```diff
@@ -15,8 +15,8 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
```

Comment:
LGTM, only question is if we do need to rename the file at all? i.e. just have `docs/build` instead of `docs/build_docs.py`

Reply:
Yes. There are several reasons for it:

1) there was a problem when I switched to Docker-build only tests in #10368 . In .dockerignore we are ignoring anything with "build" name at any level as build artifact. Usually generated directories (including one generated by setup.py create directories which are named "build". I could probably filter it out, but I think generally naming anything just "build" is a dangerous one precisely for the reason that various tools might skip or ignore it or delete it.

2) Without the .py extension this file does not get recognized as python file by pre-commit. This is why I had to fix a number of pylint/flake/mypy inside it it because it was not verified before.

3) I think generally speaking running it via buid.py should be discouraged in favour of `./breeze build-doc`. The problem is that you never know if you have all the recent extensions installed. For example if you'd do it after adding spellchecking it would have failed and you would have to figure out that you have to re-rerun 'pip install .[doc]' again. For people who are casual contributors and do it very rarely this might come as a surprise. Also they might not remember that they have a separate virtualenv where they should build the docs and they would loose time trying to find out what's going on. They also might have an even older version of the virtualenv that will not work well. 

The`./breeze build-doc` - handles all the venv problems for you and will ask you "rebuild the image" in such case,
 which makes it obvious what you should do.

Actually - it's even worse now @kaxil @mik-laj  -  I just checked that now. And the old instructions are wrong. `pip install -e '.[doc]'` does not work at all:

If you do it from the scratch you get the below error. So unless we know how to fix it, I am removing the old mechanism and replace it with breeze-only. I see no point keeping the instructions which do not work.

```
 ./docs/build_docs.py   
Recreated content of the $/home/jarek/code/airflow/docs/_build and $/home/jarek/code/airflow/docs/_api folders
Executing cmd:  sphinx-build -W -b spelling -d _build/doctrees -D extensions=sphinxarg.ext,autoapi.extension,sphinxcontrib.spelling,exampleinclude,sphinx.ext.autodoc,sphinx.ext.coverage,sphinx.ext.viewcode,sphinx.ext.graphviz,sphinxcontrib.httpdomain,sphinxcontrib.jinja,docroles,removemarktransform . _build/spelling
Running Sphinx v3.2.1
Initializing Spelling Checker 5.2.1
/home/jarek/.pyenv/versions/3.6.10/lib/python3.6/importlib/__init__.py:126: RemovedInSphinx40Warning: The alias 'sphinx.ext.autodoc.importer.mock' is deprecated, use 'sphinx.ext.autodoc.mock.mock' instead. Check CHANGES for Sphinx API modifications.
  return _bootstrap._gcd_import(name[level:], package, level)
making output directory... failed

Exception occurred:
  File "/home/jarek/.pyenv/versions/3.6.10/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '_build/spelling'
The full traceback has been saved in /tmp/sphinx-err-pj53rpvx.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
==================== Error   1 ====================
Sphinx spellcheck returned non-zero exit status: 2.

==================================================
```

Reply:
I pasted a wrong (previous output) currently You have a lot of problems like that ^^ unless you have the [al] extras installed . I really think breeze is the only option for contributors to build the docs properly.

Reply:
> For example if you'd do it after adding spellchecking it would have failed and you would have to figure out that you have to re-rerun 'pip install .[doc]' again. For people who are casual contributors and do it very rarely this might come as a surprise. Also they might not remember that they have a separate virtualenv where they should build the docs and they would loose time trying to find out what's going on. They also might have an even older version of the virtualenv that will not work well.

This is an issue that occurs with every Python project, so I don't see a big problem here. If you are missing the library, you have to reinstall the dependencies. On other hand, building documentation without Docker is faster.